### DISCUSSION TITLE
Github package registry authentication

### DISCUSSION THREAD

**Discussion by nkve** [community/community#25404] [Actions]
<p>I’m using a bespoke dependency management tool that uses the Aether library to manage maven dependencies.</p>
<p>I’d like to integrate support for deploying to github package registry, but I’m not sure what authentication information is required from github.</p>
<p>I assume it’s a matter of setting the right fields on anAether Authentication object.</p>
<p>From looking at what information maven requires I see an entry in the settings.xml</p>
<pre><code class="lang-xml">&lt;server&gt;
    &lt;id&gt;github&lt;/id&gt;
    &lt;username&gt;username&lt;/username&gt;
    &lt;password&gt;1111111111&lt;/password&gt;
&lt;/server&gt;
</code></pre>
<p>Which suggests I only need to set strings “username” and “password”</p>
<p>Despite having set these fields (along with the correct url) I keep getting an error similar to the following</p>
<pre><code>org.eclipse.aether.deployment.DeploymentException: Failed to deploy artifacts: Could not transfer artifact group:a:jar:1.0 from/to github (https://maven.pkg.github.com/user/repo): maven.pkg.github.com failed to respond
</code></pre>
<p>Any guidance would be much appreciated.</p>

**Comment by clarkbw** [comment, marked as answer]
<p>I’m not sure what that error is exactly.  For the server config we recommend using env variables as seen in <a href="https://github.com/actions/setup-java#publishing-using-apache-maven" rel="nofollow noopener">setup-java</a>. Where the password is a <a href="https://github.com/settings/tokens" rel="nofollow noopener">PAT</a> and the username is your github username.</p>
<pre><code class="lang-auto">&lt;servers&gt;
    &lt;server&gt;
      &lt;id&gt;github&lt;/id&gt;
      &lt;username&gt;${env.GITHUB_ACTOR}&lt;/username&gt;
      &lt;password&gt;${env.GITHUB_TOKEN}&lt;/password&gt;
    &lt;/server&gt;
&lt;/servers&gt;
</code></pre>
